source/util/Retry.cpp

```diff
@@ -71,4 +71,74 @@ bool Retry::exponentialBackoff(
     }
 
     return successful;
-}
\ No newline at end of file
+}
+
+bool Retry::conditionalExponentialBackoff(
+    const ExponentialRetryConfig &config,
+    const function<RetryResult_t()> &retryableFunction,
+    const function<void()> &onComplete)
```

Comment:
It looks like you don't use this 3rd parameter, so you could just delete it and simplify the logic

Reply:
The function uses the same parameters as Retry::exponentialBackoff(), so even though I'm not using onComplete when I call Retry::conditionalExponentialBackoff() I want to keep the functions having the same overall functionality